src/models/PrimaryAuth.js

```diff
@@ -106,12 +106,19 @@ function (Okta, BaseLoginModel, CookieUtil, Enums) {
 
       var primaryAuthPromise;
 
-      if (this.appState.get('isUnauthenticated') && !this.settings.get('features.passwordlessAuth')) {
-        primaryAuthPromise = this.doTransaction(function (transaction) {
-          var authClient = this.appState.settings.authClient;
-          return this.doPrimaryAuth(authClient, signInArgs, transaction.authenticate);
-        });
-      } else {
+      if (this.appState.get('isUnauthenticated')) {
+        var authClient = this.appState.settings.authClient;
+        // bootstrapped with stateToken
+        if (!this.settings.get('features.passwordlessAuth')) {
```

Comment:
I feel like this can be simplified... seems to me conditions for passwordless are not needed
Currently you have
```
if not passwordless
  -> transaction.authenticate
else if idxToken
  -> transaction.login
else
  -> transaction.authenticate
```
can we avoid checking for passwordless? Seems that it is irrelevant when we are in unauthenticated state.
```
if idxToken
 -> transaction.login
else
 -> transaction.authenticate
```

Reply:
that should work too !